src/ejabberd_sm.erl

```diff
@@ -971,7 +971,7 @@ check_max_sessions(HostType, LUser, LServer, ReplacedPIDs) ->
       Result :: infinity | pos_integer().
 get_max_user_sessions(HostType, LUser, LServer) ->
     JID = jid:make_noprep(LUser, LServer, <<>>),
-    case acl:match_rule_for_host_type(HostType, LServer, max_user_sessions, JID) of
+    case acl:match_rule(HostType, LServer, max_user_sessions, JID) of
         Max when is_integer(Max) -> Max;
```

Comment:
This just makes me realise, I think we're not checking that this number is strictly positive, right? 🤔

Reply:
We don't, that's right, it might be good to fix next.

Regarding the '++' on host and global rules, I am rewriting this right now so that it will be precomputed per host type. Expect a PR today 🙂